### PR TITLE
Move test targets information into own CMakeListst.txt file

### PR DESCRIPTION
**Pull request by slurps-mad-rips**
This makes it easier to manage the tests, and to merge in test or benchmark related changes without affecting the main cmake build itself.

This means that a future more up to date version of #1886 is more viable. 

This PR *does not* currently try to tackle the adding of benchmarks. That can be tackled after this is merged.

@saghul @bnoordhuis I don't expect this to be too big of a PR. I'd like to keep this one relegated to the tests being moved over into their own CMakeLists.txt file. I have other PRs I'd like to work on, but those are less disruptive if/when this gets merged :)

(Also, I need to make a slight edit. Two extra diffs on lines 266 and 277 snuck in. That'll be in a follow up commit once I have a moment, but I wanted to open this now for more general feedback)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -111,8 +111,9 @@ if(WIN32)
        src/win/util.c
        src/win/winapi.c
        src/win/winsock.c)
-  list(APPEND uv_test_libraries ws2_32)
-  list(APPEND uv_test_sources src/win/snprintf.c test/runner-win.c)
+  list(APPEND uv_test_sources
+    ${PROJECT_SOURCE_DIR}/src/win/snprintf.c
+    ${PROJECT_SOURCE_DIR}/test/runner-win.c)
```
Comment (bnoordhuis) [inline CMakeLists.txt]: Can you line up the arguments?

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+  test-udp-open.c
+  test-udp-options.c
+  test-udp-send-and-recv.c
+  test-udp-send-hang-loop.c
+  test-udp-send-immediate.c
+  test-udp-send-unreachable.c
+  test-udp-try-send.c
+  test-uname.c
+  test-walk-handles.c
+  test-watcher-cross-stop.c)
```
Comment (bnoordhuis) [inline test/CMakeLists.txt]: Likewise and can you drop the empty line at the bottom?

Comment (slurps-mad-rips) [reply test/CMakeLists.txt]: This'll be in the next commit :)

File: CMakeLists.txt (hunk 1 above)
Comment (slurps-mad-rips) [reply CMakeLists.txt]: Sure thing!

Comment (slurps-mad-rips) [reply CMakeLists.txt]: Actually on second thought, I'll place these in a generator expression inside the `test/CMakeLists.txt` file. Then we won't need this at all

**Comment by slurps-mad-rips** [comment]
@saghul @bnoordhuis any other quests or is this good to merge in?

**Comment by bnoordhuis** [comment]
@slurps-mad-rips https://ci.nodejs.org/job/libuv-test-commit/1760/ - it looks like this PR breaks the build. Can you take a look? (That CI run is today's HEAD of v1.x with your changes on top.)

**Comment by slurps-mad-rips** [comment]
Looks like it's a jenkins related issue. CMake knows the tests have been moved, but jenkins does not 😬
I'll see if I can get to it sometime this weekend.

**Comment by bnoordhuis** [comment]
For reference, the build error:
<details>

```
07:33:44 + /home/iojs/build/workspace/libuv-test-commit-linux-cmake/nodes/centos7-64/cmake-bin/bin/cmake -DBUILD_TESTING=ON /home/iojs/build/workspace/libuv-test-commit-linux-cmake/nodes/centos7-64
07:33:44 -- The C compiler identification is GNU 4.8.5
07:33:44 -- Check for working C compiler: /usr/lib64/ccache/cc
07:33:44 -- Check for working C compiler: /usr/lib64/ccache/cc -- works
07:33:44 -- Detecting C compiler ABI info
07:33:44 -- Detecting C compiler ABI info - done
07:33:44 -- Detecting C compile features
07:33:45 -- Detecting C compile features - done
07:33:45 -- Performing Test UV_LINT_W4
07:33:45 -- Performing Test UV_LINT_W4 - Failed
07:33:45 -- Performing Test UV_LINT_WALL
07:33:45 -- Performing Test UV_LINT_WALL - Success
07:33:45 -- Performing Test UV_LINT_NO_UNUSED_PARAMETER
07:33:45 -- Performing Test UV_LINT_NO_UNUSED_PARAMETER - Success
07:33:45 -- Performing Test UV_LINT_STRICT_PROTOTYPES
07:33:45 -- Performing Test UV_LINT_STRICT_PROTOTYPES - Success
07:33:45 -- Performing Test UV_LINT_EXTRA
07:33:45 -- Performing Test UV_LINT_EXTRA - Success
07:33:45 -- Configuring done
07:33:45 CMake Error at test/CMakeLists.txt:2 (add_executable):
07:33:45   Cannot find source file:
07:33:45 
07:33:45     test/runner-unix.c
07:33:45 
07:33:45   Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
07:33:45   .hpp .hxx .in .txx
```

</details>

It seems to just be missing runner-unix.c from the SOURCES list. Adding that (bnoordhuis/libuv@fb50bf2e) fixes the build: https://ci.nodejs.org/job/libuv-test-commit/1772/

It still needs a rebase against v1.x though, and it looks like CI adjustments on Windows are necessary because it's looking for (and not finding) `Debug\uv_run_tests.exe`.

**Comment by slurps-mad-rips** [comment]
I'll be getting back to this soon. Apologies for the delay. 😅

**Comment by slurps-mad-rips** [comment]
not stale 😬, just swamped 😅

**Comment by slurps-mad-rips** [comment]
Sorry to keep bumping this. It's not stale, but I will get to it sometime next week, hopefully :sweat_smile:

**Comment by slurps-mad-rips** [comment]
@bnoordhuis @saghul This has changed so much since I first opened it, I'm going to close this PR and try again with a clean room implementation. Apologies for not actually touching this for a few months.

**Pull request closed without merging**